Sources/NIOSSL/SSLContext.swift

```diff
@@ -159,10 +159,26 @@ public final class NIOSSLContext {
         returnCode = CNIOBoringSSL_SSL_CTX_set_cipher_list(context, configuration.cipherSuites)
         precondition(1 == returnCode)
 
+        // On non-Linux platforms, when using the platform default trust roots, we make use of a
+        // custom verify callback. If we have also been presented with additional trust roots of
+        // type `.file`, we take the opportunity now to load them in memory to avoid doing so
+        // repeatedly on the request path.
+        //
+        // However, to avoid closely coupling this code with other parts (e.g. the platform-specific
+        // concerns, and the defauling of `trustRoots` to `.default` when `nil`), we unilaterally
+        // convert any `additionalTrustRoots` of type `.file` to `.certificates`.
+        var configuration = configuration
+        configuration.additionalTrustRoots = try configuration.additionalTrustRoots.map { trustRoots in
+            guard case .file(let path) = trustRoots else { return trustRoots }
```

Comment:
This might be cleaner as a `switch` rather than a `guard case`.

Reply:
OK, will do!

Reply:
Addressed in https://github.com/apple/swift-nio-ssl/pull/284/commits/a9c3ba7a8d20f4b91be33bd53b8f3a816d751c62.